### PR TITLE
Adding support for Windows

### PR DESCRIPTION
**Pull request by angelrti**
this PR is for #3

### CODE REVIEW

File: plugins/adapters/modbus/srcCxx/ModbusAdapterConfiguration.cxx [hunk 1]
```diff
@@ -411,14 +411,20 @@ std::vector<long double> ModbusAdapterConfigurationElement::get_float_value(
             // cast the value into signed/unsigned depending on
             // the datatype used
             if (dynamic_data::is_signed_kind(element_kind)) {
-                int8_t value = input[i];
-                check_correct_value(value, index, element_kind);
+                int8_t value = static_cast<int8_t>(input[i]);
+                check_correct_value(
+                        (long double)((int8_t) value),
```
Comment (fgaranda) [inline plugins/adapters/modbus/srcCxx/ModbusAdapterConfiguration.cxx:416]: static_cast<int8_t>(value)

**Review by fgaranda — CHANGES_REQUESTED** [review]
@angelrti I've just seen a couple of C-style casts that could be replaced with `static_cast`s. Other than that, everything looks good.

One thing I've realized is that the project assumes C99 support. Should we set this explicitly this requirement using CMake using `CMAKE_C_STANDARD`? Maybe that'd preclude us from building in VS < 2015, I don't know.

**Comment by angelrti** [comment]
> @angelrti I've just seen a couple of C-style casts that could be replaced with `static_cast`s. Other than that, everything looks good.
> 
> One thing I've realized is that the project assumes C99 support. Should we set this explicitly this requirement using CMake using `CMAKE_C_STANDARD`? Maybe that'd preclude us from building in VS < 2015, I don't know.

We already set `CMAKE_C_STANDARD` to use C99 in the main CMakeLists https://github.com/rticommunity/rticonnextdds-gateway/blob/master/CMakeLists.txt

File: plugins/adapters/modbus/srcCxx/ModbusAdapterConfiguration.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelrti) [reply plugins/adapters/modbus/srcCxx/ModbusAdapterConfiguration.cxx]: done

**Review by fgaranda — APPROVED** [review]

**Pull request merged**